lib/_http_client.js

```diff
@@ -433,17 +458,15 @@ function socketOnEnd() {
   const req = this._httpMessage;
   const parser = this.parser;
 
-  if (!req.res && !req.socket._hadError) {
-    // If we don't have a response then we know that the socket
-    // ended prematurely and we need to emit an error on the request.
-    req.socket._hadError = true;
-    req.emit('error', connResetException('socket hang up'));
-  }
   if (parser) {
     parser.finish();
     freeParser(parser, req, socket);
   }
-  socket.destroy();
+
+  // If we don't have a response then we know that the socket
+  // ended prematurely and we need to emit an error on the request.
+  const err = !req.res ? connResetException('socket hang up') : null;
```

Comment:
The original check also add a `!req.socket._hadError`. Why was it removed?

Reply:
`_hadError` is not needed anymore since  error handling through stream and `errorEmitted` is used.